core/src/main/resources/hudson/widgets/HistoryWidget/index.jelly

```diff
@@ -90,5 +90,11 @@ THE SOFTWARE.
     </l:pane>
   </div>
 
+  <script defer="true">
+    // The value for `page-next-build` is modified inside of `entries.jelly` on render, so set the attribute
+    // on element `#buildHistoryPage` after that component has been rendered to get
+    // the correct value to use in `filter-build-history.js`
+    document.getElementById("buildHistoryPage").setAttribute("page-next-build", "${it.nextBuildNumberToFetch ?: it.owner.nextBuildNumber}");
```

Comment:
Would it be possible to do this without an inline script block?

Reply:
I'm not too sure sadly. Is there a way of sending the Jelly variable to `filter-build-history.js` without it?

Reply:
Add your value to a `data-whatever` attribute on an element with ID and read it from there:
https://www.jenkins.io/doc/developer/security/xss-prevention/#the-good-way

Reply:
I could add an additional element (after `<st:include page="entries.jelly" it="${page}" />` so that the value is correct) with `page-next-build` set instead of the inline JS - 

```
<st:include page="entries.jelly" it="${page}" />
<div id="properties" page-next-build="${it.nextBuildNumberToFetch ?: it.owner.nextBuildNumber}"></div>
```

If that's what you mean?

Reply:
Yes, exactly. Thanks!

(Just to be safe, CC @Wadeck )

Reply:
I've updated it to be like that :) No more inline JS 🎉